parser/validate.go

```diff
@@ -101,30 +138,17 @@ func validateMessage(res *result, syntax syntaxType, name protoreflect.FullName,
 	if index, err := internal.FindOption(res, handler, scope, md.Options.GetUninterpretedOption(), "map_entry"); err != nil {
 		return err
 	} else if index >= 0 {
-		opt := md.Options.UninterpretedOption[index]
-		optn := res.OptionNode(opt)
-		md.Options.UninterpretedOption = internal.RemoveOption(md.Options.UninterpretedOption, index)
-		valid := false
-		if opt.IdentifierValue != nil {
-			if opt.GetIdentifierValue() == "true" {
```

Comment:
Previously, this would _allow_ an explicit `map_entry` option as long as the value was false. But the latest version of `protoc` doesn't allow this at all, so we mirror its behavior.